Source/Core/VideoCommon/ShaderGenCommon.h

```diff
@@ -62,6 +59,9 @@ class ShaderGeneratorInterface
 	 */
 	template<class uid_data>
 	uid_data* GetUidData() { return nullptr; }
+
+protected:
+	std::string m_buffer;
```

Comment:
I'm quite sure an empty string won't have adverse effects. Anyhow, it's kind of odd to utilize a class intended for generating/writing shader code as the base for UID generation as well.